src/scripts/dungeons/Dungeon.ts

```diff
@@ -100,6 +120,21 @@ class Dungeon {
         return list.filter(Dungeon.isPokemon);
     }
 
+    /**
+     * Retreives the weights for all the possible enemies
+     */
+    get weightList(): number[] {
+        return this.enemyList.map((enemy) => {
+            if (typeof enemy === 'string') {
+                return 1;
+            } else if (enemy.hasOwnProperty('pokemon')) {
+                return (<DetailedPokemon>enemy).options.weight ?? 1;
+            } else {
+                return (<DungeonTrainer>enemy).options?.weight ?? 1;
+            }
+        });
+    }
```

Comment:
@HLXII 
Now that the standard dungeon enemies can have requirements, should we be checking those here too?

Reply:
Updated to handle those (though there aren't any use cases yet, so it isn't tested).

You'd probably also need to update `getPokemonDungeons` to handle the requirements like `getPokemonBossDungeons` for the bot as well.